File: scripts/package/disease_gene_omim.py
from __future__ import annotations

from pathlib import Path

from rdflib import URIRef

from utils.log_util import get_logger
from package.rdf_build_support import load_config
from package.disease_gene_association_util import (
    GENCC_SOURCE_URI,
    MIM,
    load_gencc_definitive_associations,
    load_omim_gene_associations,
    merge_association_maps,
    write_gene_association_ttl,
)

logger = get_logger()


def disease_gene_omim() -> None:
    config = load_config("config.ini")

    omim_ncbi_gene_map = load_omim_gene_associations(config["medgen_mim2gene_path"])
    print(f"OMIM_NCBIGene All Count : {len(omim_ncbi_gene_map)}")

    gencc_associations = load_gencc_definitive_associations(
        config["ncbigene_file_path"],
        config["mondo_owl_path"],
        config["gencc_submissions_path"],
    )
    before_merge = len(omim_ncbi_gene_map)
    merge_association_maps(omim_ncbi_gene_map, gencc_associations.omim_associations)
    print(f"GenCC_ncbigene_omim Count : {len(omim_ncbi_gene_map) - before_merge}")

    source_uri_map = {
        "MedGen": URIRef("ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/mim2gene_medgen"),
        "GenCC": URIRef(GENCC_SOURCE_URI),
    }
    write_gene_association_ttl(
        output_path=Path(config["rdf_output_dir"]) / "OMIM_Gene_Association.ttl",
        associations=omim_ncbi_gene_map,
        disease_context_prefix="OMIM",
        disease_namespace_prefix="mim",
        disease_namespace=MIM,
        disease_id_prefix="",
        source_uri_map=source_uri_map,
    )


if __name__ == "__main__":
    disease_gene_omim()

File: scripts/java/DiseaseGeneOMIM.java
package pubcasefinder_260415;

import java.io.IOException;
import java.util.LinkedHashMap;
import java.util.LinkedHashSet;

// MedGen 기반 OMIM 질환-유전자 연관에 GenCC를 보강해 RDF를 생성한다.
public class DiseaseGeneOMIM {
	/** MedGen과 GenCC를 병합해 OMIM_Gene_Association.ttl을 생성한다. */
	public static void main(String[] args) throws IOException {

		LinkedHashMap<String, LinkedHashSet<String>> omimNcbiGeneMap =
				DiseaseGeneAssociationUtil.loadOmimGeneAssociations(DiseaseGeneAssociationUtil.MEDGEN_MIM2GENE_PATH);
		System.out.println("OMIM_NCBIGene All Count : " + omimNcbiGeneMap.size());

		try {
			DiseaseGeneAssociationUtil.GenCCAssociations genccAssociations = DiseaseGeneAssociationUtil.loadGenccDefinitiveAssociations();
			int beforeMerge = omimNcbiGeneMap.size();
			DiseaseGeneAssociationUtil.mergeAssociationMaps(omimNcbiGeneMap, genccAssociations.omimAssociations);
			System.out.println("GenCC_ncbigene_omim Count : " + (omimNcbiGeneMap.size() - beforeMerge));
		}
		catch (Exception e) {
			throw new IOException("Failed to load GenCC associations", e);
		}

		LinkedHashMap<String, String> sourceUriMap = RdfBuildSupport.createStringMap(
				"MedGen", "ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/mim2gene_medgen",
				"GenCC", DiseaseGeneAssociationUtil.GENCC_SOURCE_URI);

		DiseaseGeneAssociationUtil.writeGeneAssociationTtl(
				DiseaseGeneAssociationUtil.RDF_DIR + "/OMIM_Gene_Association.ttl",
				omimNcbiGeneMap,
				"OMIM",
				"mim:",
				"PREFIX mim: <https://omim.org/entry/>",
				sourceUriMap);
	}
}
